packages/payment/src/services/payment.ts

```diff
@@ -0,0 +1,161 @@
+  }
+>(Payment)<TEntity> {
+  protected captureRepository_: DAL.RepositoryService
+  protected refundRepository_: DAL.RepositoryService
+  protected paymentRepository_: DAL.RepositoryService
+
+  constructor(container: InjectedDependencies) {
+    // @ts-ignore
+    super(...arguments)
+
+    this.paymentRepository_ = container.paymentRepository
+    this.captureRepository_ = container.captureRepository
+    this.refundRepository_ = container.refundRepository
+  }
+
+  /**
+   * NOTE TODO - TEMP IMPL. THAT WILL CHANGE
+   */
+  protected transformQueryForAmounts(config: FindConfig<TEntity>) {
+    const select = config?.select
+    let relations = config?.relations
+    const amountsSelect: string[] = []
+
+    if (select?.includes("captured_amount")) {
+      amountsSelect.push("captured_amount")
+      if (!relations?.includes("captures")) {
+        if (!relations) {
+          relations = ["captures"]
+        } else {
+          relations?.push("captures")
+        }
+      }
+    }
+
+    if (select?.includes("refunded_amount")) {
+      amountsSelect.push("refunded_amount")
+      if (!relations?.includes("refunds")) {
+        if (!relations) {
+          relations = ["refunds"]
+        } else {
+          relations?.push("refunds")
+        }
+      }
+    }
+
+    return { select, relations, amountsSelect }
+  }
+
+  /**
+   * NOTE TODO - TEMP IMPL. THAT WILL CHANGE
+   */
+  protected decorateAmounts(payment: TEntity, amountsSelected: string[]) {
+    const amounts = {}
+
+    for (const amountName of amountsSelected) {
+      if (amountName === "captured_amount") {
+        amounts["captured_amount"] = payment.captures.reduce(
+          (sum, capture) => sum + Number(capture.amount),
+          0
+        )
+      }
+      if (amountName === "refunded_amount") {
+        amounts["refunded_amount"] = payment.refunds.reduce(
+          (sum, refund) => sum + Number(refund.amount),
+          0
+        )
+      }
+    }
+
+    return Object.assign(payment, amounts) as TEntity
+  }
+
+  /**
+   * NOTE TODO - TEMP IMPL. THAT WILL CHANGE
+   */
+  @InjectManager("paymentRepository_")
+  async retrieve(
+    id: string,
+    config: FindConfig<any> = {}, // TODO: fix type when overriding
+    @MedusaContext() sharedContext?: Context
+  ): Promise<TEntity> {
+    const { select, relations, amountsSelect } =
+      this.transformQueryForAmounts(config)
+
+    const result = await super.retrieve(
+      id,
+      { ...config, select, relations },
+      sharedContext
+    )
+
+    return this.decorateAmounts(result, amountsSelect)
+  }
+
+  @InjectTransactionManager("captureRepository_")
+  async capture(
+    data: CreateCaptureDTO,
+    @MedusaContext() sharedContext?: Context
+  ): Promise<CaptureDTO> {
+    const created = await this.captureRepository_.create(
+      [
+        {
+          payment: data.payment_id,
+          ...data,
+        },
+      ],
+      sharedContext
+    )
+
+    return created[0]
```

Comment:
`capture` and `refund` could accept bulk operations. Or create a method with a different signature like `captureBulk`, where it is returned a tuple of all captured payments and all errors.

Reply:
just to confirm, payment providers are not yet implemented, are they?

Reply:
Sounds good, I will add bulk methods as well.

Yes - providers are not yet implemented